tests/test_bgp_af.rb

```diff
@@ -106,407 +329,55 @@ def test_collection_not_empty
-    afs.each do |af|
-      advertise_l2vpn_evpn(55, 'red', af)
-    end
-  end
-
-  ##
-  ## get_dampen_igp_metric
-  ##
-  def test_dampen_igp_metric
-    asn = '44'
-    vrf = 'green'
-    af = %w(ipv4 multicast)
-
-    bgp_af = RouterBgpAF.new(asn, vrf, af)
-
-    #
-    # Verify default value
-    #
-    assert_equal(bgp_af.dampen_igp_metric, bgp_af.default_dampen_igp_metric,
-                 "Error: Default 'dampen-igp-metric' value should be " \
-                   "#{bgp_af.default_dampen_igp_metric}")
-
-    #
-    # Set and verify 'dampen-igp-metric <value>'
-    #
-
-    # Do a 'dampen-igp-metric 555'
-    pattern = /^ *dampen-igp-metric 555$/
-    bgp_af.dampen_igp_metric = 555
-
-    af_string = get_bgp_af_cfg(asn, vrf, af)
-
-    assert_match(pattern, af_string,
-                 "Error: 'dampen-igp-metric 555' is not configured " \
-                   'and should be')
-    #
-    # Test getter
-    #
-    assert_equal(bgp_af.dampen_igp_metric, 555,
-                 'Error: dampen_igp_metric should be 555')
-
-    #
-    # Set and verify 'no dampen-igp-metric'
-    #
-
-    # Do a 'no dampen-igp-metric'
-    pattern = /no dampen-igp-metric$/
-    bgp_af.dampen_igp_metric = nil
-
-    af_string = get_bgp_af_cfg(asn, vrf, af)
-
-    assert_match(pattern, af_string,
-                 "Error: 'no dampen-igp-metric' is not configured " \
-                   'and should be')
-    #
-    # Test getter
-    #
-    assert_equal(bgp_af.dampen_igp_metric, nil,
-                 'Error: dampen_igp_metric should be nil')
-
-    #
-    # Set default value explicitly
-    #
-    bgp_af.dampen_igp_metric = bgp_af.default_dampen_igp_metric
-    assert_equal(bgp_af.dampen_igp_metric, bgp_af.default_dampen_igp_metric,
-                 "Error: Default 'dampen-igp-metric' value should be " \
-                   "#{bgp_af.default_dampen_igp_metric}")
-  end
+  ########################################################
+  #                      PROPERTIES                      #
+  ########################################################
 
-  ##
-  ## dampening
-  ##
-  # rubocop:disable Metrics/MethodLength
   def test_dampening
     asn = '101'
-
-    ############################################
-    # Set and verify 'dampening' with defaults #
-    ############################################
-    vrf = 'orange'
+    vrf = 'default'
+    vrf = 'orange' if platform == :nexus
     af = %w(ipv4 unicast)
+    config_ios_xr_dependencies(asn, vrf) if platform == :ios_xr
     bgp_af = RouterBgpAF.new(asn, vrf, af)
 
     # Test no dampening configured
     assert_nil(bgp_af.dampening)
 
-    pattern = /^ *dampening$/
-
+    ############################################
+    # Set and verify 'dampening' with defaults #
+    ############################################
     bgp_af.dampening = []
+    assert_equal(bgp_af.default_dampening,
+                 bgp_af.dampening) if platform == :nexus
+    assert_equal('', bgp_af.dampening) if platform == :ios_xr
```

Comment:
is [] not the default on XR? if so then I'd prefer you update the `bgp_af.dampening = []` above since the comment indicates that we're setting to default.